src/clr-angular/data/datagrid/render/header-renderer.ts

```diff
@@ -4,16 +4,33 @@
  * The full license information can be found in LICENSE in the root directory of this project.
  */
 import { Directive, ElementRef, EventEmitter, OnDestroy, Output, Renderer2 } from '@angular/core';
-import { Subscription } from 'rxjs';
+import { BehaviorSubject, Subscription } from 'rxjs';
 
 import { DomAdapter } from '../../../utils/dom-adapter/dom-adapter';
 import { DatagridRenderStep } from '../enums/render-step.enum';
 import { ColumnResizerService } from '../providers/column-resizer.service';
 import { STRICT_WIDTH_CLASS } from './constants';
 import { DatagridRenderOrganizer } from './render-organizer';
+import { DatagridColumnState } from '../interfaces/column-state.interface';
+import { DatagridColumnChanges } from '../enums/column-changes.enum';
 
 @Directive({ selector: 'clr-dg-column', providers: [ColumnResizerService] })
 export class DatagridHeaderRenderer implements OnDestroy {
+  stateSubscription: Subscription;
+  _columnState: BehaviorSubject<DatagridColumnState>;
+
+  get columnState(): BehaviorSubject<DatagridColumnState> {
+    return this._columnState;
+  }
+
+  set columnState(columnState: BehaviorSubject<DatagridColumnState>) {
```

Comment:
Can you add a comment that we should figure out a way to avoid the code duplication between `DatagridCellRenderer` and `DatagridHeaderRenderer` in the future? They started with a few lines of common code, but it's now grown to dozens.

Not something we have to do now, but a `TODO` comment lets us remember it next time we add another feature in here.